app/routes/_layout+/bookings.update-existing.tsx

```diff
@@ -0,0 +1,473 @@
+
+import {
+  data,
+  error,
+  getCurrentSearchParams,
+  parseData,
+} from "~/utils/http.server";
+import { getParamsValues } from "~/utils/list";
+import {
+  PermissionAction,
+  PermissionEntity,
+} from "~/utils/permissions/permission.data";
+import { requirePermission } from "~/utils/roles.server";
+export async function loader({ context, request }: LoaderFunctionArgs) {
+  const searchParams = getCurrentSearchParams(request);
+  const ids = searchParams.getAll("id");
+  const authSession = context.getSession();
+  const { userId } = authSession;
+  try {
+    const { organizationId, isSelfServiceOrBase } = await requirePermission({
+      userId: authSession?.userId,
+      request,
+      entity: PermissionEntity.booking,
+      action: PermissionAction.create,
+    });
+
+    const searchParams = getCurrentSearchParams(request);
+    const { page, search, tab } = getParamsValues(searchParams);
+    const perPage = 20;
+    const { bookings, bookingCount } = await getBookings({
+      organizationId,
+      page,
+      perPage,
+      search,
+      userId: authSession?.userId,
+      statuses: ["DRAFT", "RESERVED"],
+      ...(isSelfServiceOrBase && {
+        // If the user is self service, we only show bookings that belong to that user)
+        custodianUserId: authSession?.userId,
+      }),
+    });
+
+    const header: HeaderData = {
+      title: "Bookings",
+    };
+
+    const modelName = {
+      singular: "booking",
+      plural: "bookings",
+    };
+
+    const totalPages = Math.ceil(bookingCount / perPage);
+
+    const hints = getClientHint(request);
+
+    const items = bookings.map((b) => {
+      if (b.from && b.to) {
+        const from = new Date(b.from);
+        const displayFrom = getDateTimeFormat(request, {
+          dateStyle: "short",
+          timeStyle: "short",
+        }).format(from);
+
+        const to = new Date(b.to);
+        const displayTo = getDateTimeFormat(request, {
+          dateStyle: "short",
+          timeStyle: "short",
+        }).format(to);
+
+        return {
+          ...b,
+          displayFrom: displayFrom.split(","),
+          displayTo: displayTo.split(","),
+          metadata: {
+            ...b,
+            displayFrom: displayFrom.split(","),
+            displayTo: displayTo.split(","),
+          },
+        };
+      }
+      return b;
+    });
+    return json(
+      data({
+        showModal: true,
+        header,
+        bookings: items,
+        search,
+        page,
+        bookingCount: bookingCount,
+        totalPages,
+        perPage,
+        modelName,
+        ids: ids.length ? ids : undefined,
+        tab,
+        hints,
+      }),
+      {
+        headers: [
+          setCookie(await setSelectedOrganizationIdCookie(organizationId)),
+        ],
+      }
+    );
+  } catch (cause) {
+    const reason = makeShelfError(cause, { userId });
+    throw json(error(reason), { status: reason.status });
+  }
+}
+
+const getBookingDetails = async (
```

Comment:
I see you added 4 functions in this file that I feel are just repeating existing stuff and can be managed way better.

1. They should not be in this file. Each function should be in the relevant service file for the model it refers to
2. All function should have a try/catch.